src/components/Notification/Notification.tsx

```diff
@@ -198,15 +205,21 @@ const Notification = ({
           {hasActions ? (
             <div className="p-notification__actions">
               {actions.map((action, i) => (
-                <Button
-                  appearance={ButtonAppearance.LINK}
-                  className="p-notification__action"
-                  data-testid="notification-action"
-                  key={`${action.label}-${i}`}
-                  onClick={action.onClick}
-                >
-                  {action.label}
-                </Button>
+                <Fragment key={i}>
+                  {action && typeof action === "object" && "label" in action ? (
+                    <Button
+                      type="button"
+                      appearance={ButtonAppearance.LINK}
+                      className="p-notification__action"
+                      data-testid="notification-action"
+                      onClick={action.onClick}
+                    >
+                      {action.label}
+                    </Button>
+                  ) : (
+                    isValidElement(action) && action
+                  )}
```

Comment:
I think you can swap this around to make it simpler if you use the `isReactNode` typeguard from src/utils.tsx. It'd probably also be a good idea to add a unit test for the ReactNode case.

```suggestion
                  {isReactNode(action) ? (
                    action
                  ) : (
                    <Button
                      type="button"
                      appearance={ButtonAppearance.LINK}
                      className="p-notification__action"
                      data-testid="notification-action"
                      onClick={action.onClick}
                    >
                      {action.label}
                    </Button>
                  )}
```

Reply:
Done, thanks!